modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryMetricsTests.java

```diff
@@ -302,4 +382,30 @@ public HttpHandler getDelegate() {
             return delegate;
         }
     }
+
+    static class S3ErrorResponse {
```

Comment:
nit: looks like a `record` to me?

Reply:
Fixed in c579c56325d6f27d670897fdcf41578e0ae45554